### PR TITLE
support session parameters

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
Dumpling doesn't support customized session variables now.

### What is changed and how it works?
Add `params` argument to support session params in dumping progress. For example, `--params "character_set_client=latin1,character_set_connection=latin1"`

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
  Manually set `--params` and it looks good.

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 
### Release note

- support --params argument

### CODE REVIEW

File: cmd/dumpling/main.go [hunk 1]
```diff
@@ -127,6 +128,7 @@ func main() {
 	pflag.StringVar(&csvDelimiter, "csv-delimiter", "\"", "The delimiter for values in csv files, default '\"'")
 	pflag.StringVar(&outputFilenameFormat, "output-filename-template", "", "The output filename template (without file extension)")
 	pflag.BoolVar(&completeInsert, "complete-insert", false, "Use complete INSERT statements that include column names")
+	pflag.StringToStringVar(&sessionParams, "params", nil, `Extra session params used in dumping process, accept format: --params "character_set_client=latin1,character_set_connection=latin1"`)
```
Comment (kennytm) [inline cmd/dumpling/main.go:131]:
```suggestion
	pflag.StringToStringVar(&sessionParams, "params", nil, `Extra session variables used while dumping, accepted format: --params "character_set_client=latin1,character_set_connection=latin1"`)
```

**Review by kennytm — COMMENTED** [review]
Rest LGTM (calling them "session variables" to reuse the same terminology with MySQL).

File: cmd/dumpling/main.go (hunk 1 above)
Comment (lichunzhu) [reply cmd/dumpling/main.go]: ok. Do we need to change the argument name `params` to `session-variables` or something else?

Comment (kennytm) [reply cmd/dumpling/main.go]: i think it's fine.

Comment (lichunzhu) [reply cmd/dumpling/main.go]: OK. Some other code has been changed since the last commit. PTAL again.

File: v4/export/sql.go [hunk 2]
```diff
@@ -481,13 +481,7 @@ func resetDBWithSessionParams(db *sql.DB, dsn string, params map[string]interfac
...
 
 	for k, v := range support {
-		var s string
-		if str, ok := v.(string); ok {
-			s = wrapStringWith(str, "'")
-		} else {
-			s = fmt.Sprintf("%v", v)
-		}
-		dsn += fmt.Sprintf("&%s=%s", k, url.QueryEscape(s))
+		dsn += fmt.Sprintf("&%s=%s", k, url.QueryEscape(fmt.Sprintf("%v", v)))
```
Comment (kennytm) [inline v4/export/sql.go:484]: the `if` which was introduced in #111, it this now irrelevant with `--snapshot`?

Comment (lichunzhu) [reply v4/export/sql.go]: This commit has been reverted. I add two comments here.

**Review by kennytm — APPROVED** [review]
LGTM

**Pull request merged**